scripts/us_census/acs5yr/subject_tables/common/generate_col_map.py

```diff
@@ -0,0 +1,284 @@
+            if 'ignoreColumns' in self.features and col not in self.features[
+                    'ignoreColumns']:
+                try:
+                    self.column_map[col] = self._column_to_statvar(col)
+                except Exception as e:
+                    exec_type, exec_obj, exec_tb = sys.exc_info()
+                    logger.error(
+                        f"Exeception: {exec_type} occured in generate_col_map.py at line {exec_tb.tb_lineno}"
+                    )
+                    continue
+
+        #TODO: Enable validating the generate columnMap. Dependent implementation _isvalid_column_map
+        return self.column_map
+
+    def _column_to_statvar(self, column):
+        """generates a dictionary statistical variable with all properties specified in the JSON spec for a single column"""
+        measurement_assigned = False
+        stat_var = {}
+        # part_list contains a list of tokens of the column name after splitting the
+        # column name based on the '!!' delimiter
+        part_list = column.split(self.delimiter)
+
+        # part_list[-1], the last element of part_list describes the property value.
+        # This element points to the 'v' in 'pv' -> and we map it to the
+        # corresponding property 'p'
+        # Eg:
+        # 'Estimate!!Total!!Total civilian noninstitutionalized
+        # population!!SEX!!Male' is the name of the column and the value you we want
+        # is the gender 'Male'. This pattern repeats for all columns
+        # To verify this assumption for specific census tables, please check the
+        # table shells at: https://www2.census.gov/programs-surveys/acs/tech_docs/table_shells/2019/
+        val_prop_col = part_list[-1]
+
+        for part in part_list:
+            # set the base for special cases like median, etc.
+            if not measurement_assigned and part in self.features['measurement']:
+                stat_var.update(self.features['measurement'][part])
+                measurement_assigned = True
+
+        # set the default statVar definition
+        if not measurement_assigned and '_DEFAULT' in self.features[
+                'measurement']:
+            stat_var.update(self.features['measurement']['_DEFAULT'])
+
+        # set the populationType attribute
+        if 'populationType' not in stat_var:
+            stat_var['populationType'] = self._get_population_type(part_list)
+
+        # associate pvs to stat_var
+        #p = property, k = dictionary of substrings in column that points to a propertyValue
+        for p, k in self.features['pvs'].items():
+            for c, v in k.items():
+                ## check if the current key of dict matches with any substring from the tokens of a column name
+                for part in part_list:
+                    # check if column substring matches with a column token
+                    # the check is done by ignoring the case
+                    if c.lower() in part.lower():
```

Comment:
I'm not sure I can find explanation you refer to in https://github.com/datacommonsorg/data/pull/488#discussion_r699690780.  I'm still not sure why we won't do exact column match part match!